DigitalLearningSolutions.Data/Services/DelegateUploadFileService.cs

```diff
@@ -0,0 +1,337 @@
+                "Answer6",
+                "Active",
+                "EmailAddress"
+            }.OrderBy(x => x);
+            var actualHeaders = table.Fields.Select(x => x.Name).OrderBy(x => x);
+            return actualHeaders.SequenceEqual(expectedHeaders);
+        }
+
+        private bool? TryGetExistingApprovedStatus(string? delegateId, string? aliasId, int centreId)
+        {
+            if (!string.IsNullOrWhiteSpace(delegateId))
+            {
+                var approvedStatus = userDataService.GetApprovedStatusFromCandidateNumber(delegateId, centreId);
+                if (!approvedStatus.HasValue)
+                {
+                    throw new UserAccountNotFoundException(string.Empty);
+                }
+
+                return approvedStatus.Value;
+            }
+
+            return !string.IsNullOrWhiteSpace(aliasId)
+                ? userDataService.GetApprovedStatusFromAliasId(aliasId, centreId)
+                : null;
+        }
+
+        public class DelegateTableRow
+        {
+            public DelegateTableRow(IXLTable table, IXLRangeRow row)
+            {
+                string? FindFieldValue(string name)
+                {
+                    var col = table.FindColumn(col => col.FirstCell().Value.ToString() == name).ColumnNumber();
+                    return row.Cell(col).GetValue<string?>();
+                }
+
+                RowNumber = row.RowNumber();
+                DelegateId = FindFieldValue("DelegateID");
+                LastName = FindFieldValue("LastName");
+                FirstName = FindFieldValue("FirstName");
+                JobGroupId = FindFieldValue("JobGroupID");
+                Active = FindFieldValue("Active");
+                Answer1 = FindFieldValue("Answer1");
+                Answer2 = FindFieldValue("Answer2");
+                Answer3 = FindFieldValue("Answer3");
+                Answer4 = FindFieldValue("Answer4");
+                Answer5 = FindFieldValue("Answer5");
+                Answer6 = FindFieldValue("Answer6");
+                AliasId = FindFieldValue("AliasID");
+                Email = FindFieldValue("EmailAddress");
+            }
+
+            public int RowNumber { get; set; }
+            public string? DelegateId { get; set; }
+            public string? FirstName { get; set; }
+            public string? LastName { get; set; }
+            public string? JobGroupId { get; set; }
+            public string? Active { get; set; }
+            public string? Answer1 { get; set; }
+            public string? Answer2 { get; set; }
+            public string? Answer3 { get; set; }
+            public string? Answer4 { get; set; }
+            public string? Answer5 { get; set; }
+            public string? Answer6 { get; set; }
+            public string? AliasId { get; set; }
+            public string? Email { get; set; }
+
+            public BulkUploadResult.ErrorReasons? ValidateFields(IEnumerable<int> allowedJobGroupIds)

```

Comment:
Not super keen on the fact that the validation is split across various places. Might be nice if we could bring them together.